api/messages.proto

```diff
@@ -34,10 +35,28 @@ message Ticket {
 
   // Properties contains custom info about the ticket.  Top level values can be
   // used in indexing and filtering to find tickets.
+  // TODO: Deprecate and remove this field.
   google.protobuf.Struct properties = 2;
 
   // Assignment associated with the Ticket.
   Assignment assignment = 3;
+
+  SearchFields search_fields = 4;
+
+  google.protobuf.Any criteria = 5;
+}
+
+// Search fields are the fields which Open Match is aware of, and can be used
+// when specifying filters.
+message SearchFields {
+  // Float arguments.  Filterable on ranges.
+  map<string, double> fargs = 1;
+ 
+  // String arguments.  Filterable on equality.
+  map<string, string> sargs = 2;
+ 
+  // Filterable on presence or absence of given value.
```

Comment:
examples on each of these would be super useful - definitely for this one. may be just my familiarity with old model but stating this is a bool filter seems to clarify well for me.

Reply:
Added examples to the filter specification, where the others examples are as well.